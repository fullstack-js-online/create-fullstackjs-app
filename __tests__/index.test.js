const sao = require('sao')
const path = require('path')
const generator = path.join(__dirname, '..')

const defaultPromptAnswers = {
    server: 'express',
    database: 'mongodb',
    features: ['prettier', 'linter'],
    pm: 'yarn',
    client: 'react',
    test: 'jest'
}

test('reactjs', async () => {
    const mockPromptAnswers = Object.assign({}, defaultPromptAnswers, { client: 'react' })

    const stream = await sao.mock({ generator }, mockPromptAnswers)

    expect(stream.fileList).toMatchSnapshot()
})

test('vue js', async () => {
    const mockPromptAnswers = Object.assign({}, defaultPromptAnswers, { client: 'vue' })

    const stream = await sao.mock({ generator }, mockPromptAnswers)

    expect(stream.fileList).toMatchSnapshot()
})

test('mysql', async () => {
    const mockPromptAnswers = Object.assign({}, defaultPromptAnswers, { database: 'mysql' })

    const stream = await sao.mock({ generator }, mockPromptAnswers)

    expect(stream.fileList).toMatchSnapshot()
})
