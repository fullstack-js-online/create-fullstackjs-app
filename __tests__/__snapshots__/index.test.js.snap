// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mysql 1`] = `
Array [
  ".DS_Store",
  ".babelrc",
  ".env",
  ".env.example",
  ".eslintrc.json",
  ".gitignore",
  ".prettierignore",
  ".prettierrc",
  "README.md",
  "__mocks__/@fullstackjs/mail.js",
  "__tests__/client/components/button.spec.js",
  "__tests__/server/functional/index.spec.js",
  "__tests__/setup/client.js",
  "client/components/Button/Button.js",
  "client/components/Button/index.js",
  "client/components/EmailConfirmationLoader/EmailConfirmationLoader.js",
  "client/components/EmailConfirmationLoader/index.js",
  "client/components/Flash/Flash.js",
  "client/components/Flash/index.js",
  "client/components/ForgotPasswordForm/ForgotPasswordForm.js",
  "client/components/ForgotPasswordForm/index.js",
  "client/components/Hero/Hero.js",
  "client/components/Hero/index.js",
  "client/components/Loader/Loader.js",
  "client/components/Loader/index.js",
  "client/components/LoginForm/LoginForm.js",
  "client/components/LoginForm/index.js",
  "client/components/Logo/Logo.js",
  "client/components/Logo/index.js",
  "client/components/PageLoader/PageLoader.js",
  "client/components/PageLoader/index.js",
  "client/components/RegisterForm/RegisterForm.js",
  "client/components/RegisterForm/index.js",
  "client/components/ResetPasswordForm/ResetPasswordForm.js",
  "client/components/ResetPasswordForm/index.js",
  "client/components/SendEmailConfirm/SendEmailConfirm.js",
  "client/components/SendEmailConfirm/index.js",
  "client/components/Textbox/Textbox.js",
  "client/components/Textbox/index.js",
  "client/components/Topbar/Topbar.js",
  "client/components/Topbar/index.js",
  "client/index.js",
  "client/pages/EmailConfirmation/EmailConfirmation.js",
  "client/pages/EmailConfirmation/index.js",
  "client/pages/ForgotPassword/ForgotPassword.js",
  "client/pages/ForgotPassword/index.js",
  "client/pages/Home/Home.js",
  "client/pages/Home/index.js",
  "client/pages/Login/Login.js",
  "client/pages/Login/index.js",
  "client/pages/Main/Main.js",
  "client/pages/Main/index.js",
  "client/pages/Register/Register.js",
  "client/pages/Register/index.js",
  "client/pages/ResetPassword/ResetPassword.js",
  "client/pages/ResetPassword/index.js",
  "client/store/actions/auth.js",
  "client/store/index.js",
  "client/store/reducers/auth.js",
  "client/store/reducers/index.js",
  "client/styles/main.css",
  "client/validation-schemas/index.js",
  "jest.config.js",
  "knexfile.js",
  "mail.config.js",
  "nodemon.json",
  "package.json",
  "postcss.config.js",
  "server/app.js",
  "server/config/index.js",
  "server/controllers/v1/auth.controller.js",
  "server/index.js",
  "server/mails/confirm-email/confirm-email.html.hbs",
  "server/mails/confirm-email/confirm-email.text.hbs",
  "server/mails/forgot-password/forgot-password.html.hbs",
  "server/mails/forgot-password/forgot-password.text.hbs",
  "server/middleware/auth.js",
  "server/migrations/20190319002236_create_users_table.js",
  "server/migrations/20190319004159_create_password_resets_table.js",
  "server/models/User.js",
  "server/public/.gitignore",
  "server/public/index.html",
  "server/public/loading.dark.png",
  "server/public/loading.png",
  "server/public/normalize.css",
  "server/routes/index.js",
  "server/routes/v1/auth.js",
  "server/validation-schemas/index.js",
  "server/validators/email-confirmation.js",
  "server/validators/forgot-password.js",
  "server/validators/login.js",
  "server/validators/register.js",
  "server/validators/reset-password.js",
  "tailwind.js",
  "webpack.config.js",
]
`;

exports[`reactjs 1`] = `
Array [
  ".DS_Store",
  ".babelrc",
  ".env",
  ".env.example",
  ".eslintrc.json",
  ".gitignore",
  ".prettierignore",
  ".prettierrc",
  "README.md",
  "__mocks__/@fullstackjs/mail.js",
  "__tests__/client/components/button.spec.js",
  "__tests__/server/functional/index.spec.js",
  "__tests__/setup/client.js",
  "client/components/Button/Button.js",
  "client/components/Button/index.js",
  "client/components/EmailConfirmationLoader/EmailConfirmationLoader.js",
  "client/components/EmailConfirmationLoader/index.js",
  "client/components/Flash/Flash.js",
  "client/components/Flash/index.js",
  "client/components/ForgotPasswordForm/ForgotPasswordForm.js",
  "client/components/ForgotPasswordForm/index.js",
  "client/components/Hero/Hero.js",
  "client/components/Hero/index.js",
  "client/components/Loader/Loader.js",
  "client/components/Loader/index.js",
  "client/components/LoginForm/LoginForm.js",
  "client/components/LoginForm/index.js",
  "client/components/Logo/Logo.js",
  "client/components/Logo/index.js",
  "client/components/PageLoader/PageLoader.js",
  "client/components/PageLoader/index.js",
  "client/components/RegisterForm/RegisterForm.js",
  "client/components/RegisterForm/index.js",
  "client/components/ResetPasswordForm/ResetPasswordForm.js",
  "client/components/ResetPasswordForm/index.js",
  "client/components/SendEmailConfirm/SendEmailConfirm.js",
  "client/components/SendEmailConfirm/index.js",
  "client/components/Textbox/Textbox.js",
  "client/components/Textbox/index.js",
  "client/components/Topbar/Topbar.js",
  "client/components/Topbar/index.js",
  "client/index.js",
  "client/pages/EmailConfirmation/EmailConfirmation.js",
  "client/pages/EmailConfirmation/index.js",
  "client/pages/ForgotPassword/ForgotPassword.js",
  "client/pages/ForgotPassword/index.js",
  "client/pages/Home/Home.js",
  "client/pages/Home/index.js",
  "client/pages/Login/Login.js",
  "client/pages/Login/index.js",
  "client/pages/Main/Main.js",
  "client/pages/Main/index.js",
  "client/pages/Register/Register.js",
  "client/pages/Register/index.js",
  "client/pages/ResetPassword/ResetPassword.js",
  "client/pages/ResetPassword/index.js",
  "client/store/actions/auth.js",
  "client/store/index.js",
  "client/store/reducers/auth.js",
  "client/store/reducers/index.js",
  "client/styles/main.css",
  "client/validation-schemas/index.js",
  "jest.config.js",
  "mail.config.js",
  "nodemon.json",
  "package.json",
  "postcss.config.js",
  "server/.DS_Store",
  "server/app.js",
  "server/config/index.js",
  "server/controllers/auth.controller.js",
  "server/index.js",
  "server/mails/confirm-email/confirm-email.html.hbs",
  "server/mails/confirm-email/confirm-email.text.hbs",
  "server/mails/forgot-password/forgot-password.html.hbs",
  "server/mails/forgot-password/forgot-password.text.hbs",
  "server/middleware/auth.js",
  "server/models/PasswordReset.js",
  "server/models/User.js",
  "server/public/.gitignore",
  "server/public/index.html",
  "server/public/loading.dark.png",
  "server/public/loading.png",
  "server/public/normalize.css",
  "server/routes/index.js",
  "server/routes/v1/auth.js",
  "server/validation-schemas/index.js",
  "server/validators/email-confirmation.js",
  "server/validators/forgot-password.js",
  "server/validators/login.js",
  "server/validators/register.js",
  "server/validators/reset-password.js",
  "tailwind.js",
  "webpack.config.js",
]
`;

exports[`vue js 1`] = `
Array [
  ".DS_Store",
  ".babelrc",
  ".env",
  ".env.example",
  ".eslintrc.json",
  ".gitignore",
  ".prettierignore",
  ".prettierrc",
  "README.md",
  "__mocks__/@fullstackjs/mail.js",
  "__tests__/client/components/__snapshots__/button.spec.js.snap",
  "__tests__/client/components/button.spec.js",
  "__tests__/server/functional/index.spec.js",
  "client/components/Button.vue",
  "client/components/Flash.vue",
  "client/components/Loader.vue",
  "client/components/Logo.vue",
  "client/components/Textbox.vue",
  "client/components/Topbar.vue",
  "client/index.js",
  "client/mixins/auth.js",
  "client/mixins/flash.js",
  "client/mixins/form.js",
  "client/pages/EmailConfirm.vue",
  "client/pages/ForgotPassword.vue",
  "client/pages/Home.vue",
  "client/pages/Login.vue",
  "client/pages/Main.vue",
  "client/pages/Register.vue",
  "client/pages/ResetPassword.vue",
  "client/routes.js",
  "client/store/index.js",
  "client/store/modules/auth/actions.js",
  "client/store/modules/auth/getters.js",
  "client/store/modules/auth/index.js",
  "client/store/modules/auth/mutations.js",
  "client/store/modules/flash/index.js",
  "client/store/modules/flash/mutations.js",
  "client/styles/main.css",
  "client/utils/axios.js",
  "jest.config.js",
  "mail.config.js",
  "nodemon.json",
  "package.json",
  "postcss.config.js",
  "server/.DS_Store",
  "server/app.js",
  "server/config/index.js",
  "server/controllers/auth.controller.js",
  "server/index.js",
  "server/mails/confirm-email/confirm-email.html.hbs",
  "server/mails/confirm-email/confirm-email.text.hbs",
  "server/mails/forgot-password/forgot-password.html.hbs",
  "server/mails/forgot-password/forgot-password.text.hbs",
  "server/middleware/auth.js",
  "server/models/PasswordReset.js",
  "server/models/User.js",
  "server/public/.gitignore",
  "server/public/index.html",
  "server/public/loading.dark.png",
  "server/public/loading.png",
  "server/public/normalize.css",
  "server/routes/index.js",
  "server/routes/v1/auth.js",
  "server/validation-schemas/index.js",
  "server/validators/email-confirmation.js",
  "server/validators/forgot-password.js",
  "server/validators/login.js",
  "server/validators/register.js",
  "server/validators/reset-password.js",
  "tailwind.js",
  "webpack.config.js",
]
`;
